frontend/src/lib/RunUtils.ts

```diff
@@ -66,6 +66,32 @@ function getPipelineName(run?: ApiRun | ApiJob): string | null {
   return (run && run.pipeline_spec && run.pipeline_spec.pipeline_name) || null;
 }
 
+function getPipelineVersionId(run?: ApiRun | ApiJob): string | null {
+  return run &&
+    run.resource_references &&
+    run.resource_references.some(
+      ref => ref.key && ref.key.type && ref.key.type === ApiResourceType.PIPELINEVERSION,
+    )
+    ? run.resource_references.find(
+        ref => ref.key && ref.key.type && ref.key.type === ApiResourceType.PIPELINEVERSION,
+      )!.key!.id!
```

Comment:
Recommend being safe and not use !, provide a default value if key or id is null

Reply:
Actually, the 3 !s have been checked by (1) some (2) ref.key (3) ref.key.id respectively. So the all !s just to take the value and otherwise, if any fails, it goes to null branch.

Reply:
SG, let me introduce optional chaining, so it won't be so painful later